compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -175,10 +175,26 @@ impl<'a> AstValidator<'a> {
                     }
                 }
             }
+            TyKind::AnonymousStruct(ref fields, ..) | TyKind::AnonymousUnion(ref fields, ..) => {
+                self.with_banned_assoc_ty_bound(|this| {
+                    walk_list!(this, visit_struct_field_def, fields)
+                });
+            }
             _ => visit::walk_ty(self, t),
         }
     }
 
+    fn visit_struct_field_def(&mut self, field: &'a FieldDef) {
```

Comment:
```suggestion
    fn check_struct_field_def(&mut self, field: &'a FieldDef) {
```
Could you avoid naming inherent methods with `visit_*`?
It makes them confusable with overridable `Visitor` methods.

(Same applies to `fn visit_ty_common`.)

Reply:
Hmm, "check" seems like it's validating something, which isn't entirely true. Maybe "traverse" would be a better name. what do you think?

Reply:
Maybe "walk" like `walk_ty` nearby.

Reply:
But `walk_ty` mirrors `visit::walk_ty`, so that would imply `walk_struct_field_def` exists on the `Visitor` trait.

Reply:
Ok, I don't really care much.